megamek/src/megamek/client/ui/advancedsearch/AdvancedSearchEquipmentClass.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.advancedsearch;
+
+import java.util.Locale;
+
+enum AdvancedSearchEquipmentClass {
```

Comment:
Could you build the list by the Flags?  For Example Watchdog doesn't have Probe or ECM but has these flags. 

or(F_ECM).or(F_BAP)

Reply:
no. we are searching the cache entries, therefore we have only the names.